OPHD/States/MapViewStateHelper.cpp

```diff
@@ -596,16 +596,18 @@ NAS2D::Xml::XmlElement* writeResearch(const ResearchTracker& tracker)
 
 void readResearch(NAS2D::Xml::XmlElement* element, ResearchTracker& tracker)
 {
-	const std::vector<std::string> researchList = NAS2D::split(element->attribute("completed"));
+	if (!element) { return; }
+	
+	const std::vector<std::string> researchList = NAS2D::split(element->attribute("completed_techs"));
 
 	for (auto& item : researchList)
 	{
 		tracker.addCompletedResearch(std::stoi(item));
 	}
 
 	for (auto currentResearch = element->firstChildElement();
-		currentResearch != nullptr;
-		currentResearch = currentResearch->nextSiblingElement())
+		 currentResearch != nullptr;
+		 currentResearch = currentResearch->nextSiblingElement())
```

Comment:
The indentation here looks suspect. I think it may be mixing tabs and spaces.

Reply:
Interesting. Will investigate.